transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/XmlResponseQueueTransformer.java

```diff
@@ -10,42 +10,354 @@
+                        } catch (CatalogTransformerException | IOException e) {
+                            cancelOperation.set(true);
+                            throw new RuntimeException("Failure to write node; operation aborted",
+                                    e);
+                        }
+                    }
+                }
+                writer.endNode(); // metacard
+            }
+            writer.flush();
+            return stringWriter;
+        }
+
+        private void writeAttributeToXml(MetacardPrintWriter writer, Attribute attribute,
+                AttributeFormat format) throws IOException, CatalogTransformerException {
+            String attributeName = attribute.getName();
+
+            for (Serializable value : attribute.getValues()) {
+                String xmlValue = null;
+
+                switch (format) {
+
+                case STRING:
+                case BOOLEAN:
+                case SHORT:
+                case INTEGER:
+                case LONG:
+                case FLOAT:
+                case DOUBLE:
+                    xmlValue = value.toString();
+                    break;
+                case DATE:
+                    Date date = (Date) value;
+                    xmlValue = DateFormatUtils.formatUTC(date, DF_PATTERN);
+                    break;
+                case GEOMETRY:
+                    xmlValue = geoToXml(new GeometryTransformer().transform(attribute));
+                    break;
+                case OBJECT:
+                    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                    try (ObjectOutput output = new ObjectOutputStream(bos)) {
+                        output.writeObject(attribute.getValue());
+                        xmlValue = Base64.encode(bos.toByteArray());
+                    }
+                    break;
+                case BINARY:
+                    xmlValue = Base64.encode((byte[]) value);
+                    break;
+                case XML:
+                    xmlValue = value.toString().replaceAll("[<][?]xml.*[?][>]", "");
+                    break;
+                }
+
+                // Write the node if we were able to convert it.
+                if (xmlValue != null) {
+                    writer.startNode(TYPE_NAME_LOOKUP.get(format));
+                    writer.addAttribute("name", attributeName);
+                    writer.startNode("value");
+
+                    if (format == AttributeFormat.XML || format == AttributeFormat.GEOMETRY) {
+                        writer.setRawValue(xmlValue);
+                    } else {
+                        writer.setValue(xmlValue);
+                    }
+
+                    writer.endNode(); // value
+                    writer.endNode(); // type
+                }
+            }
+        }
+
+        private String geoToXml(BinaryContent content) {
+            try {
+                XmlPullParser parser = XppFactory.createDefaultParser();
```

Comment:
I don't know its internals, so I don't want to assume it's threadsafe. That caveat aside, I could construct one in `doCompute()` that would be thread-isolated. That's a nice catch.